dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/ReconnectTimerTask.java

```diff
@@ -41,9 +41,18 @@ protected void doTask(Channel channel) {
         try {
             Long lastRead = lastRead(channel);
             Long now = now();
+
+            // Rely on reconnect timer to reconnect when AbstractClient.doConnect fails to init the connection
+            if (!channel.isConnected()) {
```

Comment:
A channel may still keep connected but has not read/write for `n*heartbeat`, which means `channel.isConnect` may always return true. I don't see you ever try to close the channel on receiving say n times of `IdleStateEvent `.

Reply:
Never mind, it's ok to go. Now I understand that `!channel.isConnected()` here is used to guarantee initial connection.

Reply:
server will close the channel.